pkg/pinning/pinning_test.go

```diff
@@ -0,0 +1,61 @@
+package pinning
+
+import (
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/statestore/mock"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+var _ Interface = (*Service)(nil)
+
+func TestPinningService(t *testing.T) {
+	var (
+		svc  = NewService(mock.NewStateStore())
+		addr = swarm.NewAddress([]byte{0})
+	)
+
+	if err := svc.CreatePin(addr); err != nil {
+		t.Fatalf("CreatePin(...): unexpected error: %v", err)
+	}
+	// The second call is to test idempotency.
```

Comment:
but...strictly speaking you only testing that the result is as  desired after 2 calls :)